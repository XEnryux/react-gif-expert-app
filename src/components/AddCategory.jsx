import { useState } from "react";
import PropTypes from 'prop-types';


export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue ] = useState('');


//Esta configuracion de setState para el imput permite que se
//renderice lo que se va escribiendo 
//en el teclado mientras se esta escribiendo 
//dentro de la barra de busqueda

    const onImputChange = ({target}) => {   
        setInputValue(target.value)
    };


    // funcion ejecutada al enviar el formulario de la barra de busquesa
    // esta linea de codigo aniade lo escrito en la barra de busqueda al arreglo original
    // de "categories" de 'GifExpertApp
    const onSubmit = (event) => {
    
        event.preventDefault();
        if( inputValue.trim().length <= 1) return;

        // setCategories(categories => [inputValue,...categories]);
        onNewCategory(inputValue.trim());
        setInputValue('');
      

    }

    return (
        <form onSubmit= { onSubmit } aria-label = 'form'>
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onImputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,

  }

