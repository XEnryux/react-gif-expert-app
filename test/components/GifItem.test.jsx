import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"


describe('pruebas al componente <GifItem/>', () => {


    const title = "titulo"
    const url = "https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=1eac2f8c6m8gelxnx8cayxs7fcowvi673k75ikd5fpeeglom&rid=giphy.gif&ct=g"


    test('debe de hacer match con el snapshot', () => {



        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot()
    });

    test('debe de mostrar la imagen con el url y el alt indicado', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)
        const {src,alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
     });
     test('debe de mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url} />);
        expect(screen.getByAltText(title)).toBeTruthy()
      });
})