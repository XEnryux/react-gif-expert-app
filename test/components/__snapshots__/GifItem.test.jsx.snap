// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas al componente <GifItem/> debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=1eac2f8c6m8gelxnx8cayxs7fcowvi673k75ikd5fpeeglom&rid=giphy.gif&ct=g"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
